src/intel_batchbuffer.c

```diff
@@ -146,11 +146,17 @@ intel_batchbuffer_flush(struct intel_batchbuffer *batch)
     used = batch->ptr - batch->map;
 
     if (batch->intel->has_watchdog)
-        intel_batchbuffer_configure_watchdog(batch->intel->fd, batch->intel->gem_ctx_id, batch->flag);
+        intel_batchbuffer_configure_watchdog(batch->intel, batch->intel->gem_ctx_id, batch->flag);
 
-    /* FIXME: Add BSD2 ring on supported platforms */
-    if (ring_flag == I915_EXEC_BSD)
-        batch->flag |= LOCAL_I915_EXEC_BSD_RING0;
+    if (batch->intel->has_bsd2) {
+        if (ring_flag == I915_EXEC_BSD)
+            batch->flag |= LOCAL_I915_EXEC_BSD_RING0;
```

Comment:
The KMD driver will use the ping-pong mode to select the right ring if neitherLOCAL_I915_EXEC_BSD_RING0 nor LOCAL_I915_EXEC_BSD_RING1 is set.  Specifying LOCAL_I915_EXEC_BSD_RING0 will cause performance drop for multiple decoding/encoding cases.

BTW LOCAL_I915_EXEC_BSD_RING0 is equal to I915_EXEC_BSD_RING1 and LOCAL_I915_EXEC_BSD_RING1 is equal to I915_EXEC_BSD_RING2, but I915_EXEC_BSD_RING1 and I915_EXEC_BSD_RING2 were not defined in older version of libdrm, so we added LOCAL_I915_EXEC_BSD_RING0 and LOCAL_I915_EXEC_BSD_RING1 in the driver.